docs/compatibility.md

```diff
@@ -0,0 +1,71 @@
+# LiT and LND compatibility
```

Comment:
We should probably take inspiration from the updated title in #1154, i.e. `## LND Version Compatibility When Used In ***Remote Mode***`.
I'm not suggesting to copy that exactly, but clarify in the title that this applies to **Remote Mode** specifically.